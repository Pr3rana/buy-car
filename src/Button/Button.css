.default-btn{
    text-decoration: none;
    border: none;
    float: right;
    font-size: 18px;
    width: 110px;
    height: 32px;
    margin: 18px;
    font-family: Roboto;
    background-color: #ea7f28;
    border-radius: 4px;
    color: #FFFFFF;
    text-align: center;
    font-size: 14px;
    /* transition: all 0.5s; */
    cursor: pointer;
}

.default-btn:active, .default-btn:focus{
    outline:0;
    background-color: #d37324;
}