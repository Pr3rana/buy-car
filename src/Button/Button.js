import './Button.css';
const Button = ({value, click, customClass=""}) => {
    return ( 
     <button className={`default-btn ${customClass}`} onClick={click}>{value}</button>
     );
}
 
export default Button;