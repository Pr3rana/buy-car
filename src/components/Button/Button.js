import './Button.css';
const Button = ({value ="", click = ()=>{}, customClass=""}) => {
    return ( 
     <button data-testid = "default-btn" className={`default-btn ${customClass}`} onClick={click}>{value}</button>
     );
}
 
export default Button;