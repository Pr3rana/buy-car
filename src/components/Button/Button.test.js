import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import Button from './Button';

afterEach(cleanup);
describe("Button render", ()=>{
    test('Initial Render', () => {
        const { container } = render(<Button/>)
        expect(container.firstChild).toMatchSnapshot(`
          <button data-testid="default-btn" class="default-btn "></button>
        `)
    });
});

describe("Button props",()=>{
    test('Re-render the Button with different props',()=>{
        const { container, getByText } = render(<Button value={"Save"} onClick={()=>{}} />)
        expect(getByText('Save')).toBeInTheDocument()
        expect(container.firstChild).toMatchSnapshot(`
          <button data-testid="default-btn" class="default-btn ">Save</button>
        `)
        expect(screen.getByTestId('default-btn')).toHaveTextContent('Save')
    })
})

describe("Button event",()=>{
    test("test button click event", ()=>{
        // console.log = jest.fn();
        const click = jest.fn()
        render(<Button value={"Save"} onClick={click()} />)
        expect(screen.getByRole('button')).toBeInTheDocument()
        // fireEvent.click(screen.getByTestId('default-btn'))
        expect(click).toHaveBeenCalledTimes(1)
        // expect(console.log).toHaveBeenCalledWith("test btn")
    })
    
})