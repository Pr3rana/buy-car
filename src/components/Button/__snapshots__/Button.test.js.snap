// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button props Re-render the Button with different props: 
          <button data-testid="default-btn" class="default-btn ">Save</button>
         1`] = `
<button
  class="default-btn "
  data-testid="default-btn"
>
  Save
</button>
`;

exports[`Button render Initial Render: 
          <button data-testid="default-btn" class="default-btn "></button>
         1`] = `
<button
  class="default-btn "
  data-testid="default-btn"
/>
`;
